presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -388,6 +395,11 @@ else if (getParquetType(type, fileSchema, useParquetColumnNames, column, tableNa
             if (e.getClass().getSimpleName().equals("BlockMissingException")) {
                 throw new PrestoException(HIVE_MISSING_DATA, message, e);
             }
+            if (e instanceof HiddenColumnException) {
+                message = format("User does not have access to encryption key for encrypted column = %s. If returning 'null' for encrypted " +
+                        "columns is acceptable to your query, please add 'set session hive.read_null_masked_parquet_encrypted_value_enabled=true' before your query", e.getMessage());
```

Comment:
Instead of `e.getMessage()`, can we actually get the column name?

Reply:
Nice feedback. Yeah, we should use the column name.